packages/opentelemetry-core/src/trace/sampler/AlwaysSampleSampler.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Sampler, SpanContext } from '@opentelemetry/types';
+
+/** Sampler that samples every trace. */
+export class AlwaysSampleSampler implements Sampler {
```

Comment:
Both `AlwaysSampleSampler` and `NeverSampleSampler` are not needed as the same can be achieved by the standard sampler with a rate of 100% and 0% respectively.

Reply:
As per the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sampling-api.md#built-in-samplers), these are the built-in samplers.

Reply:
Could you have something like `export const ALWAYS_SAMPLER = new Sampler(0)` and `export const NEVER_SAMPLER = new Sampler(1)`?

Reply:
> As per the specs, these are the built-in samplers.

How can we make sure to provide a sane API for Node when the spec enforces a lot of weirdness and in some cases outright anti-patterns? At this stage we should be able to quickly make changes to the spec so we don't end up with a poorly designed API in the end.

Otherwise there will be a lot of rewrites needed down the line at best, and hurt adoption at worst if the project is released without addressing the API.

Reply:
Since `Sampler` has no concept of a rate, we could add a `RateSampler` instead that would handle both concerns from `Always` and `Never` plus anything in between.

Reply:
Can we change to a more simpler and more flexible `RateSampler` instead of `Always` and `Never`? Having explicit classes for specific values of a range is an API anti-pattern. Since `RateSampler` is actually a superset of both, this is still compliant with the spec.

Reply:
Thanks for the suggestion. Done, added `ProbabilitySampler` instead of `RateSampler `.  Let me know if this makes sense.